dsl_parser/constraints.py

```diff
@@ -444,9 +446,10 @@ def _validate_input_value(input_name, input_constraints, input_value,
             data_based_constraints.append(c)
             continue
         if not c.predicate(input_value):
+            msg = f'is invalid. {c.error_message}' if c.error_message \
```

Comment:
how about some kinda `error_message = c.error_message.format(value=input_value)`. Then, users can use "{value}" in their error messages